main-actions/src/main/scala/sbt/compiler/EvalReporter.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * sbt
+ * Copyright 2011 - 2018, Lightbend, Inc.
+ * Copyright 2008 - 2010, Mark Harrah
+ * Licensed under Apache License 2.0 (see LICENSE)
+ */
+
+package sbt.compiler
+
+import scala.reflect.internal.settings.MutableSettings
+import scala.reflect.internal.util.Position
+import scala.tools.nsc.Settings
+import scala.tools.nsc.reporters.{ ConsoleReporter, FilteringReporter }
+
+abstract class EvalReporter extends FilteringReporter {
```

Comment:
Could you add a small Scaladoc to document the motivation behind this class please?
Something along the lines of:

```suggestion
/** Reporter used to compile metabuild (build.sbt files) to forward error messages via BSP.
 */
abstract class EvalReporter extends FilteringReporter {
```